css-view-transitions-2/Overview.bs

```diff
@@ -627,21 +627,21 @@ When capturing the old or new state for an element, perform the following steps
 
 				1. Set |inboundTransition|’s [=ViewTransition/phase=] to "`update-callback-called`".
 
-				1. Call |onReady|.
```

Comment:
I was mistaken! We do need the callback to be invoked here. The reason being that the callback executes unload steps for the old Document and those need to be deferred until we have captured it.

Reply:
Yea they need to be in the `old state captured steps` but not in the steps we post as a task to the new doc

Reply:
Oops, yes, moved it to the "old state captured steps"

Reply:
So there's 2 options for calling "proceed with navigation":

- At the end of the rendering opportunity which we're caching, the code in this PR. This is running on the [rendering task source](https://html.spec.whatwg.org/#rendering-task-source).
- At the end of the task we post to copy all the state to the new Document here. This task is running on the [dom manipulation task source](https://html.spec.whatwg.org/#dom-manipulation-task-source).

I left a comment to understand which task source these steps should run on. But another issue I see here is, what is the ordering guarantee between these task sources? Because the task which sets up the VT object on the new Document must run before any of its rendering opportunities. If these task queues are ordered then this might be ok because we queue the task to copy the state *before* invoking "proceedWithNavigation". Please leave a note to document this assumption if so.

Since the above is quite subtle, I'd be curious on what's the advantage on invoking proceedWithNavigation sooner than the task being queued here. Doing it after initializing the VT object on the new Document seems more robust.

Reply:
"Proceed with navigation" doesn't run on a task source, as those steps don't run for a document (renderer). it invokes a callback that adds steps to the traversal queue (the browser process).

The task that runs on the new document and sets up the new VT Object is defined here, and runs on the DOM manipulation task source.

I think the issue is that there might be a rendering opportunity before this task is run. That's indeed a problem, I'll refactor it to be special steps that run when the document is activated.